src/partials/bloblang-playground.hbs

```diff
@@ -0,0 +1,238 @@
+  aceMappingEditor.session.setUseSoftTabs(true);
+
+  // Initialize output editor
+  aceOutputEditor = ace.edit("ace-output");
+  aceOutputEditor.setTheme("ace/theme/github");
+  aceOutputEditor.session.setMode("ace/mode/text"); // JSON syntax highlighting
+  aceOutputEditor.setReadOnly(true); // Make output read-only
+  aceOutputEditor.session.setTabSize(4);
+  aceOutputEditor.session.setUseSoftTabs(true);
+
+  // Initialize metadata editor
+  aceMetadataEditor = ace.edit("ace-metadata");
+  aceMetadataEditor.setTheme("ace/theme/github");
+  aceMetadataEditor.session.setMode("ace/mode/json"); // Use JSON mode for metadata
+  aceMetadataEditor.setReadOnly(true); // Make metadata read-only
+  aceMetadataEditor.session.setTabSize(4);
+  aceMetadataEditor.session.setUseSoftTabs(true);
+
+  new Choices(document.getElementById("sample-dropdown"), {
+    searchEnabled: true,
+    searchPlaceholderValue: 'Search examples',
+    placeholderValue: 'Select an example',
+    allowHTML: true
+  });
+
+  const go = new Go();
+  WebAssembly.instantiateStreaming(fetch("{{{siteRootPath}}}/blobl.wasm"), go.importObject)
+    .then((result) => {
+      go.run(result.instance);
+      restoreFromLocalStorage();
+      execute();
+    })
+    .catch(console.error);
+
+  // Handle clear button
+  document.getElementById("clear").addEventListener("click", () => {
+    aceInputEditor.setValue("");
+    aceMappingEditor.setValue("");
+    aceOutputEditor.setValue(defaultOutput, 1);
+    aceMetadataEditor.setValue(defaultMeta, 1);
+    saveToLocalStorage();
+  });
+
+  // Handle prettify button
+  document.getElementById("prettify").addEventListener("click", () => {
+    try {
+      const formatted = JSON.stringify(JSON.parse(aceInputEditor.getValue()), null, 2);
+      aceInputEditor.setValue(formatted, 1);
+      saveToLocalStorage();
+    } catch (error) {
+      aceOutputEditor.setValue("Error: Invalid JSON input", 1);
+    }
+  });
+
+  const samples = {{{page.attributes.bloblang-samples}}};
+
+  // Handle dropdown for examples
+  document.getElementById("sample-dropdown").addEventListener("change", (event) => {
+    const selectedSample = event.target.value;
+    const sample = Object.values(samples).find((sample) => sample.title === selectedSample);
+
+    if (sample) {
+      aceInputEditor.setValue(sample.input, 1);
+      aceMappingEditor.setValue(sample.mapping, 1);
+      execute();
+    }
+  });
+
+  // Save content to localStorage and execute on changes
+  [aceInputEditor, aceMappingEditor].forEach((editor) => {
+    editor.on("change", saveToLocalStorage);
+    editor.on("change", execute);
+  });
+});
+
+// Function to get input and mapping values
+function getInput() {
+  return aceInputEditor ? aceInputEditor.getValue() : "";
+}
+
+function getMapping() {
+  return aceMappingEditor ? aceMappingEditor.getValue() : "";
+}
+
+function isValidJSON(str) {
+  try {
+    JSON.parse(str);
+    return true;
+  } catch (e) {
+    return false;
```

Comment:
For the result, it's not necessary. We just check so that we can format the output as JSON. If the result is not JSON, there was an error in the input or mapping, and we display that as plain text.